arch/x86/entry.S

```diff
@@ -86,7 +95,7 @@ END_FUNC(entry_\sym)
 
 .macro interrupt_handler sym func
 ENTRY(asm_interrupt_handler_\sym)
-    enter_from_usermode
+    _from_usermode
```

Comment:
Is this always triggered in user mode? What happens if interrupted in kernel mode?

Reply:
It can be triggered in kernel mode too. The code should use irq_from_usermode here.